### PR TITLE
Send invite message as bot message, not ephemeral

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Invite messages need to be send from bot, but not as ephemeral messages.

### CODE REVIEW

File: server/api.go [hunk 1]
```diff
@@ -569,7 +569,15 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
...
 		ChannelId: stateArr[3],
 	}
-	_ = a.p.GetAPI().UpdateEphemeralPost(mmUser.Id, post)
+
+	_, appErr = a.p.GetAPI().GetPost(stateArr[2])
+	if appErr == nil {
+		_, appErr = a.p.GetAPI().UpdatePost(post)
+	} else {
+		_ = a.p.GetAPI().UpdateEphemeralPost(mmUser.Id, post)
+
```
Comment (mgdelacroix) [inline server/api.go:578]: Linter is going to complain about this whitespace

File: server/connect.go [hunk 2]
```diff
@@ -93,16 +93,20 @@ func (p *Plugin) SendInviteMessage(user *model.User, pendingSince time.Time, cur
...
 	channel, err := p.apiClient.Channel.GetDirect(user.Id, p.userID)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get bot DM channel with user_id %s", user.Id)
 	}
 	message := fmt.Sprintf("@%s, you're invited to use the MS Teams connected experience. ", user.Username)
-	p.SendConnectMessage(channel.Id, user.Id, message)
+
+	postID := model.NewId()
+	connectURL := fmt.Sprintf(p.GetURL()+"/connect?post_id=%s&channel_id=%s", postID, channel.Id)
+	connectMessage := message + " " + fmt.Sprintf("[Click here to connect your account](%s)", connectURL)
```
Comment (mgdelacroix) [inline server/connect.go:104]: nit: for clarity

```suggestion
	connectMessage := fmt.Sprintf("%s [Click here to connect your account](%s)", message, connectURL)
```

**Review by mgdelacroix — APPROVED** [review]

**Review by calebroseland — APPROVED** [review]

**Review by calebroseland — APPROVED** [review]
Working beautifully

https://github.com/mattermost/mattermost-plugin-msteams/assets/11724372/aef7e1f7-889c-45c7-823a-904bebb9aa56

**Review by mgdelacroix — APPROVED** [review]
Awesome!

**Pull request merged**